besspin/cwesEvaluation/numericErrors/sources/test_128.c

```diff
@@ -1,181 +1,166 @@
-// This test is adapted from the example code
-// cited in the SEI's CERT C Coding Standard
-// Rule MEM07-C
-
+#include "testsParameters.h"
+#include <stdio.h>
 #include <stdbool.h>
-#include <stddef.h>
-#include <stdlib.h>
 #include <stdint.h>
-#include <stdio.h>
-#include <string.h>
-
-#include "testsParameters.h"
 
 #if (defined(BESSPIN_FREERTOS) && defined(BESSPIN_FPGA))
-#include "FreeRTOS.h"
-#define FREE vPortFree
-#define MALLOC pvPortMalloc
-#else
-#define FREE free
-#define MALLOC malloc
+    #include "FreeRTOS.h"
 #endif
 
-#define NUM_OF_TEST_PARTS 2
+#define NUM_OF_TEST_PARTS 4
+
+#if __riscv_xlen == 64
+    #define UINT_MAX UINT64_MAX
+#elif __riscv_xlen == 32
+    #define UINT_MAX UINT32_MAX
+#endif
+
+enum operation {MULT, ADD};
+
+static void test_op_dtype_xlen(int part, enum operation op);
+static void test_op_dtype_8(int part, enum operation op);
+static void check_test_error(bool error, int part);
+static __attribute__((noinline)) size_t add_xlen (size_t in1, size_t in2);
+static __attribute__((noinline)) uint8_t add_8 (uint8_t in1, uint8_t in2);
+static __attribute__((noinline)) size_t mult_xlen (size_t in1, size_t in2);
+static __attribute__((noinline)) uint8_t mult_8 (uint8_t in1, uint8_t in2);
 
-void check_test_error(bool error, int part) {
+#ifdef BESSPIN_FREERTOS
+    #define NUM_OF_TEST_PARTS 8
```

Comment:
Should `NUM_OF_TEST_PARTS` be 4?  There are only 4 cases in the switch statement.

Reply:
Good catch. Thanks. It was masked because the macro is not used in C unless in case of error. And the regex matches the first instance.